airflow/api_fastapi/common/parameters.py

```diff
@@ -112,7 +112,7 @@ def depends(self, only_active: bool = True) -> _OnlyActiveFilter:
         return self.set_value(only_active)
 
 
-class _SearchParam(BaseParam[str]):
+class SearchParam(BaseParam[str]):
```

Comment:
Can we let it be a internal class and instead using directly in task_instances.py, 
we can define a query variable under parameters? 
`QueryTaskDisplayNamePatternSearch`
I think we follow the similar for other searches too.

Reply:
Hi @shubhamraj-git,  
From my perspective, I think `task_display_name_pattern` is not currently used by other endpoints, so it might not be suitable to place it in common parameters. WDYT ?
Here some related discussion https://github.com/apache/airflow/pull/43947#issuecomment-2503874473 about common parameters.

Additionally, some `_SearchParam` constructs in common parameters should be refactored out ( since those only be used in one endpoint ). It seems I overlooked this in https://github.com/apache/airflow/pull/43947 PR.

Reply:
I could see us using `task_display_name_pattern` in other endpoints.

Reply:
If then, I will move the it to common parameter. Thanks for catching @bbovenzi !